### PR TITLE
[Tester] allow running single tests with 'run' option

### PR DESCRIPTION
**Pull request by Wh1teDuke**
Because of how was defined previosly, it was impossible to run single tests. It would happen that either the file would be ignored and instead the category processed or that tester would try to find the test file inside tests folder.

Now `koch test run async/tupcoming_async` does actually work.

### CODE REVIEW

File: tests/testament/categories.nim [hunk 1]
```diff
@@ -376,7 +376,19 @@ proc `&?.`(a, b: string): string =
...
-proc processCategory(r: var TResults, cat: Category, options: string, fileGlob: string = "t*.nim") =
+proc processCategory(
+    r: var TResults, cat: Category, options: string, fileGlob: string = "") =
+  
+  if fileGlob != "":
+    # run option
+    let testFile = "tests" & DirSep &.? cat.string / fileGlob
+  
+    if not existsFile(testFile & ".nim"):
+      raise newException(AssertionError, testFile & " test does not exist")
```
Comment (Araq) [inline tests/testament/categories.nim]: That's not how to handle user input errors.

**Comment by Araq** [comment]
Seems convoluted for what it does.

**Comment by Wh1teDuke** [comment]
Unless you want to remove the alert altogether, i can't think of a way to make it less convoluted.

**Comment by Araq** [comment]
Well for a start, categories should be prioritized over single tests and we could require the '.nim' extension for single tests. This should simplify the logic already.

**Comment by Wh1teDuke** [comment]
>Well for a start, categories should be prioritized over single tests

That's the behaviour for ['all' and 'category'](https://github.com/nim-lang/Nim/blob/devel/tests/testament/tester.nim#L437-L443) option, which calls 'processCategory' without the file argument. But because of how was programmed, it affected 'run' too.

>we could require the '.nim' extension for single tests

I'll change this.

**Comment by Wh1teDuke** [comment]
Nevermind, i will add a new 'processSingleTest' proc.

**Pull request closed without merging**